### DISCUSSION TITLE
Pass URL parameters via QRCode to fill in a form in a screen

### DISCUSSION THREAD

**Discussion by bmscmoreira** [Budibase/budibase#12323] [Help] [answered]
Is this workflow possible in Budibase?

- user scans a QRCode which has a link to budibase app and url parameters;
- the budibase app opens in a specific screen which has a form and pre-populates the forms values based on the url parameters passed by the link / QR Code

Thanks

**Comment by andz-bb** [comment]
hey @bmscmoreira 

I just tried this flow in cloud with multiple URL parameters and it works well. If you scan the code on an unauthenticated device it will redirect you to the exact URL on the QR after signing in.

Here's a quick demo I made with a public screen on my cloud app:

![qr-code](https://github.com/Budibase/budibase/assets/110921612/f5102813-b9b3-44c0-9bda-cf9b2b3276b6)

As you can see, some of the form fields are autopopulated with values from URL parameters included in my QR code.

Hope this helps!

**Reply by bmscmoreira** [reply]
Great! Just tried it, it works! Mind to share how to configure your screen / form to accept those parameters?

**Reply by andz-bb** [reply, marked as answer]
Sure, [the docs explains it in detail on this page.](https://docs.budibase.com/docs/url-parameters)

The quick TLDR version is:

- update your screen's route to include `/:variable_name` for each parameter you want to include
- use the `{{ URL.variable_name }}` binding as the default value for each relevant field
- navigate to the screen route replacing `:variable_name` with the desired value
- if you're using special characters in the parameter value, you will have to encode the value so that it is URL safe

<img width="1218" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/1daf43ee-3ccd-436d-951a-5328d23e8311">

hope this helps - if you're having any problems with the setup let me know!

**Reply by bmscmoreira** [reply]
Thank you. Still getting my hands on all these concepts.
